### PR TITLE
Quote bot

### PR DESCRIPTION
**Pull request by laiqjafri**
# Quote Bot #
It quotes a famous quote of a dialogue from a movie every 10 minutes.
If mentioned in a tweet, it will mention your name in a quote.
If direct messaged, it will reply with a quote

You will need to set the following ENV variables:
1. TWITTER\_CONSUMER\_KEY
2. TWITTER\_CONSUMER\_SECRET
3. TWITTER\_ACCESS\_TOKEN
4. TWITTER\_ACCESS\_TOKEN\_SECRET
5. [MASHAPE_KEY](http://mashape.com/andruxnet/random-famous-quotes)

### CODE REVIEW

File: app/models/quote.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Quote
+  attr_accessor :text, :author, :category
+
+  def initialize options = {}
+    self.text     = options[:text]     || options["text"] || options[:quote] || options["quote"]
```
Comment (lcpriest) [inline app/models/quote.rb]: You could normalize this, right? options.symbolize_keys!

Comment (laiqjafri) [inline app/models/quote.rb]: Yes, I should have used this method.

File: app/models/quote_engine.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class QuoteEngine
+  QUOTES_API_URL = "https://andruxnet-random-famous-quotes.p.mashape.com/cat=random"
+  def self.random
+    response = Unirest.post QUOTES_API_URL, headers: headers
+    return Quote.new(response.body) if response.code == 200
+    return Quote.new(:text => "Seems like I am out of any new quotes", :author => 'GeckoBot', :category => 'Wrong')
```
Comment (lcpriest) [inline app/models/quote_engine.rb]: hash syntax

Comment (lcpriest) [inline app/models/quote_engine.rb]: { text: "Seems like I am out of any new quotes" }

Comment (laiqjafri) [inline app/models/quote_engine.rb]: yep, got it

**Pull request closed without merging**